documentation/gcp-account-and-identity-architecture.md

```diff
@@ -0,0 +1,105 @@
+encourage automated configurations delivery and operation to production. The DevOps team has no plans to enforce that.
+
+### UC2 - Provision Accounts
+
+The HelpfulEng DevOps team will provision GCP accounts for both shared and project delivery accounts. The DevOps team
+would like provision these accounts in a standardized way with low effort and simple adoption of Cloud security and
+governance practices.
+
+## Logical Architecture
+
+This design provisions a set of GCP accounts for each project team to deliver their applications and a few shared
+accounts for prototyping, security, and governance. The Helpful Engineering GCP Organization will look like: 
+
+```
+helpfulengineering.org Org
+└ Core
+  └ Log
+  └ Audit
+└ Sandboxes
+  └ helpful-gcp-sandbox2
+└ Project Delivery
+  └ Monitoring O2
+    └ helpful-project-monitoring-O2-dev
+    └ helpful-project-monitoring-O2-prod
+  └ <project name>
+    └ helpful-<project name>-dev
+    └ helpful-<project name>-prod
+  └ ...
+```
+
+The design accommodates the wide set of people, skills, and techniques used throughout Helpful Engineering and scopes the
+management, fault, and security domain to the project team.  Using separate GCP accounts for each project team 
+creates a strong security boundary between teams that isolates each project's activities, resources, and
+data from each other.  
+
+The Helpful Engineering GCP Organization will be configured to match the depicted hierarchy. This Organization structure enables use of both shared and project-specific Service Control Policy.
+
+The DevOps team will use [GCP Cloud Identity](https://cloud.google.com/identity/docs) 
+to provision both the shared and project accounts. Control Tower provides a simple account provisioning model that
+provides a number of security and governance best practices out of the box. 
+
+**Core Accounts**
+
+The `Log` account contains the organization's API activity logs [Cloud Audit Logs](https://cloud.google.com/logging/docs/audit) and resource configurations (Config).
+
+The `Audit` account is a restricted account that gives security and compliance teams read and write access to all 
+accounts in the landing zone.
+
+**Sandbox Accounts**
+
+The `helpful-gcp-sandbox` account is a sandbox account teams can use to experiment and prototype solutions.  Applications should not be operated for customers out of 
+sandbox  accounts.  The DevOps team should consider enforcing this policy with a governance tool like Cloud Custodian
+that destroys resources some number of days after provisioning, e.g. 10 days.
+
+**Project Delivery Accounts**
+
+The DevOps team will provision two 'delivery' accounts for each project team: `dev` and `prod`.  Teams will use the
+`dev` account to develop their applications and test application deployments.  Applications should be delivered
+to the production account for operation and use by customers and end users.  
```

Comment:
qq: should each HE project get separate dev and prod accounts in GCP?  what advantages does this have over a dev and prod GCP project within a single GCP account for the HE project?

I definitely think we should create strong isolation between environments and projects, just wondering what the right balance is.  AFAIK, a GCP IAM principal:

* must be specifically granted access to resources in different projects
* may be granted permissions to call different API actions in different projects in a single account  (is this true?)